File: CoreJava/College-app/src/com/techlabs/college/Professor.java
package com.techlabs.college;

import java.util.*;

public class Professor extends Person implements SalariedEmployee {
	private int id;
	private String address;
	private Date dob;
	private double basic;
	private double hra;
	private double da;

	public Professor(int id, String address, Date dob, Double basic) {
		this.id = id;
		this.address = address;
		this.dob = dob;
		this.basic = basic;
		this.hra = basic / 2;
		this.da = basic/3;
	}

	public double getBasic() {
		return basic;
	}

	public double getHra() {
		return hra;
	}

	public double getDa() {
		return da;
	}

	public int getId() {
		return id;
	}

	public String getAddress() {
		return address;
	}

	public Date getDob() {
		return dob;
	}

	@Override
	public double calcSalary() {
		return getBasic() + getDa() + getHra();
	}

}

File: DotNET/C#/CollegeApp/CollegeApp/Professor.cs
using System;

namespace CollegeApp
{
    class Professor
    {
        private int _id;
        private String _address;
        private DateTime _dob;
        private double _basic;
        private double _hra;
        private double _da;

        public Professor(int id, String address, DateTime dob, double basic)
        {
            _id = id;
            _address = address;
            _dob = dob;
            _basic = basic;
            _hra = basic / 2;
            _da = basic / 3;
        }

        public String Address
        {
            get
            {
                return _address;
            }
        }

        public DateTime Dob
        {
            get
            {
                return _dob;
            }
        }

        public double Basic
        {
            get
            {
                return _basic;
            }
        }

        public double Id
        {
            get
            {
                return _id;
            }
        }

        public double Da
        {
            get
            {
                return _da;
            }
        }

        public double Hra
        {
            get
            {
                return _hra;
            }
        }

        public double CalculateSalary()
        {
            return Basic + Da + Hra;
        }

    }
}
